editor/src/clj/editor/pipeline.clj

```diff
@@ -200,7 +200,14 @@
                         (render-progress! (swap! progress progress/with-message message))
                         (let [result (or cached-artifact
                                          (let [dep-resources (make-dep-resources deps build-targets-by-content-hash)
-                                               build-result (build-fn resource dep-resources user-data)]
+                                               build-result (try
+                                                              (build-fn resource dep-resources user-data)
+                                                              (catch OutOfMemoryError error
+                                                                (g/error-aggregate
+                                                                  [(g/error-fatal
+                                                                     (format "Failed to allocate memory while building '%s': %s."
+                                                                             (resource/proj-path resource)
+                                                                             (.getMessage error)))])))]
```

Comment:
Report out-of-memory exceptions from the `build-fn` to the Build Errors tab instead of showing an exception dialog. This could maybe help our users work around the issue in some situations (i.e. split up a huge atlas).